src/main/java/com/microsoft/lst_bench/common/LSTBenchmarkExecutor.java

```diff
@@ -141,6 +140,7 @@ public void execute() throws Exception {
             new ObjectMapper().writeValueAsString(experimentMetadata));
         throw e;
       } finally {
+        executor.shutdownNow();
```

Comment:
Great catch!

I think current solution will not work when you have multiple repetitions for the test, since once the method `shutdown` is called in the pool, it will not accept new tasks. Unfortunately, we do not have tests with multiple repetitions, which is why CI did not catch it.

One possible solution is rather than moving the whole thread pool initialization code block, keeping that inside the loop as it was, but moving the reference outside the loop (i.e., `ExecutorService executor = null;`). Then in the finally block, do a `null` check on the executor for safety, and proceed to call `shutdown` if it is not. Iirc the reason to have `shutdown` + waiting time was to give opportunity to threads to push any remaining telemetry to the registry, thus maybe those two lines (L125-126) should be simply moved to L143? I'm not sure they make much sense where they are now, maybe it was an oversight during refactoring of the old code.

On another note, would you mind creating a GH issue to include tests with multiple repetitions? That way we will not forget about it.